lib/src/platform/android/map_state.dart

```diff
@@ -533,32 +533,34 @@ final class MapLibreMapStateAndroid extends MapLibreMapStateNative {
   }
 
   @override
-  Future<void> setStyle(String style) async {
-    final trimmed = style.trim();
-    final builder = jni.Style$Builder();
-    if (trimmed.startsWith('{')) {
-      // Raw JSON
-      builder.fromJson(trimmed.toJString());
-    } else if (trimmed.startsWith('/')) {
-      builder.fromUri('file://$trimmed'.toJString());
-    } else if (!trimmed.startsWith('http://') &&
-        !trimmed.startsWith('https://') &&
-        !trimmed.startsWith('mapbox://')) {
-      // flutter asset
-      final content = await rootBundle.loadString(trimmed);
-      builder.fromJson(content.toJString());
-    } else {
-      // URI
-      builder.fromUri(trimmed.toJString());
-    }
-    _jniMapLibreMap?.setStyle$3(
-      builder,
-      jni.Style$OnStyleLoaded.implement(
-        jni.$Style$OnStyleLoaded(
-          onStyleLoaded: (style) => onStyleLoaded(),
+  void setStyle(String style) {
+    (() async {
+      final trimmed = style.trim();
+      final builder = jni.Style$Builder();
+      if (trimmed.startsWith('{') || trimmed.startsWith('[')) {
+        // Raw JSON
+        builder.fromJson(trimmed.toJString());
+      } else if (trimmed.startsWith('/')) {
+        builder.fromUri('file://$trimmed'.toJString());
+      } else if (!trimmed.startsWith('http://') &&
+          !trimmed.startsWith('https://') &&
+          !trimmed.startsWith('mapbox://')) {
+        // flutter asset
+        final content = await rootBundle.loadString(trimmed);
+        builder.fromJson(content.toJString());
+      } else {
+        // URI
+        builder.fromUri(trimmed.toJString());
+      }
+      _jniMapLibreMap?.setStyle$3(
+        builder,
+        jni.Style$OnStyleLoaded.implement(
+          jni.$Style$OnStyleLoaded(
+            onStyleLoaded: (style) => onStyleLoaded(),
+          ),
         ),
-      ),
-    );
-    builder.release();
+      );
+      builder.release();
+    })();
   }
```

Comment:
@mhernz Is there an advantage to use an anonymous function here? I'd think having the interface sync but keeping the actual implementation async is a bit more clean?

Reply:
I think I was trying to avoid the lint. I'll add an ignore comment instead